_pytest/pytester.py

```diff
@@ -418,16 +418,8 @@ class Testdir:
     def __init__(self, request, tmpdir_factory):
         self.request = request
         self._mod_collections = WeakKeyDictionary()
-        # XXX remove duplication with tmpdir plugin
-        basetmp = tmpdir_factory.ensuretemp("testdir")
```

Comment:
this removal slightly changes output structure, i advice against gutting that bit

Reply:
Hmm but as I see it this is the central change which fixes the behavior: use the same logic that `tmpdir` uses to handle this behavior.

If we fear this might break something, we should target `features` then, but the new behavior seems more correct IMO.

Reply:
But what do you suggest instead of the current code?

Reply:
Just rebased to features

Reply:
on features i consider the change fine, thanks